frontend/src/admin/categories/EditCategory.tsx

```diff
@@ -0,0 +1,174 @@
+import { useEffect, useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { Link, useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { useCookies } from 'react-cookie';
+import { useNavigate } from 'react-router-dom';
+import {
+  Nav,
+  Container,
+  FormInput,
+  Button,
+  Loader,
+  Notification,
+} from '@components/';
+import { setIsInvalid, setIsLoading, setIsError } from '@store/category';
+import { RootState } from '@store/store';
+import { getSpecificCategory, editCategory } from '@api/CategoryApi';
+import './index.css';
+
+type Inputs = {
+  name: string;
+  description: string;
+};
+
+export default function EditCategory() {
+  const state = useSelector((state: RootState) => state.category);
+  const [categoryItem, setCategoryItem] = useState({
+    id: 0,
+    name: '',
+    description: '',
+    created_at: '',
+    updated_at: '',
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [cookies] = useCookies();
+  const { category_id } = useParams();
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<Inputs>();
+
+  const categoryValidation = {
+    name: {
+      required: {
+        value: true,
+        message: 'Category name is required.',
+      },
+      onChange: () => dispatch(setIsInvalid(false)),
+    },
+    description: {
+      required: false,
+    },
+  };
+
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    dispatch(setIsLoading(true));
+    if (!data.description) {
+      data.description = 'null';
+    }
+    editCategory(cookies.admin_token, categoryItem.id.toString(), data)
+      .then((response) => {
+        dispatch(setIsLoading(false));
+        navigate('/admin/dashboard');
+      })
+      .catch((error) => {
+        if (error.response?.status === 422) {
+          dispatch(setIsInvalid(true));
+        } else {
+          console.log(error);
```

Comment:
Better to use below, pls check rest of your code
```suggestion
          console.error(error);
```

Reply:
ah i'm not using this line anymore diay miss since naa naman koy handler for errors so will remove this line